src/main/java/uk/gov/hmcts/darts/audio/service/impl/OutboundAudioDeleterProcessorImpl.java

```diff
@@ -66,15 +72,36 @@ public List<MediaRequestEntity> markForDeletion() {
 
 
         for (TransientObjectDirectoryEntity entity : transientObjectDirectoryEntities) {
-            entity.getTransformedMedia().getMediaRequest().setStatus(MediaRequestStatus.EXPIRED);
             entity.getTransformedMedia().getMediaRequest().setLastModifiedBy(systemUser);
 
             entity.setLastModifiedBy(systemUser);
             entity.setStatus(deletionStatus);
-
+            entity.getTransformedMedia().setExpiryTime(OffsetDateTime.now());
             deletedValues.add(entity.getTransformedMedia().getMediaRequest());
         }
 
+        for (Integer mediaRequestId : mediaRequests) {
```

Comment:
I'm thinking this class would be better to be rewritten, starting with line 51.
instead of getting a list of media requests that have transformed medias that are expired, i think we should be just getting a list of transformed media that are expired, then loop through that list.

using SQL that is very roughly somethgin like this:-

```
    @Query("""
        SELECT tm FROM MediaRequestEntity mr, TransformedMediaEntity tm
        WHERE tm.mediaRequest = mr
        AND ((tm.lastAccessed < :lastAccessedDateTime
        AND mr.status = "COMPLETED")
        OR (tm.createdDate < :lastAccessedDateTime
        AND mr.status <> "PROCESSING")
        AND tm.lastAccessed IS NULL)

        """)
```



The current implementation has me concerned that if a media request has 2 transformed media, and 1 of those was accessed today, and one hasn;t been accessed for a week, then this might deelte the one accessed recently also.

Reply:
ACCEPT